homeassistant/components/imap/coordinator.py

```diff
@@ -359,7 +359,7 @@ async def _cleanup(self, log_error: bool = False) -> None:
                 await self.imap_client.stop_wait_server_push()
                 await self.imap_client.close()
                 await self.imap_client.logout()
-            except (AioImapException, TimeoutError):
+            except (AioImapException, asyncio.CancelledError, TimeoutError):
```

Comment:
This will also consume legitimate attempts to cancel 

https://superfastpython.com/asyncio-cancel-task-and-wait/#Naive_Cancel_And_Wait

Reply:
Removed this one